checklist.py

```diff
@@ -450,5 +450,31 @@ def _parse_file(self, fp, root):
         return skbio_imports
 
 
+class FluxCapacitorValidator(RepoValidator):
+    """Ensure that the __future__ statements are fluxing correctly"""
+    reason = ("These files do not have the following import at the start:\n\n"
+              "from __future__ import absolute_import, division,"
+              " print_function\n")
+
+    def _validate(self, root, dirs, files):
+        failures = []
+        expected = set(("absolute_import", "division", "print_function"))
+        for file in files:
+            if file.endswith(".py"):
+                filename = os.path.join(root, file)
+                failures.append(filename)
+                with open(filename) as f:
+                    source = ast.parse(f.read())
+                    for node, _ in zip(ast.iter_child_nodes(source), range(2)):
+                        if isinstance(node, ast.Expr):
+                            continue
+                        if isinstance(node, ast.ImportFrom):
+                            if node.module == "__future__":
+                                if expected <= set(n.name for n in node.names):
```

Comment:
Thanks!

On Mon, Jun 29, 2015 at 9:30 PM, Jai Ram Rideout <notifications@github.com>
wrote:

> In checklist.py
> <https://github.com/biocore/scikit-bio/pull/971#discussion_r33537569>:
>
> > +
> > +    def _validate(self, root, dirs, files):
> > +        failures = []
> > +        expected = set(("absolute_import", "division", "print_function"))
> > +        for file in files:
> > +            if file.endswith(".py"):
> > +                filename = os.path.join(root, file)
> > +                failures.append(filename)
> > +                with open(filename) as f:
> > +                    source = ast.parse(f.read())
> > +                    for node, _ in zip(ast.iter_child_nodes(source), range(2)):
> > +                        if isinstance(node, ast.Expr):
> > +                            continue
> > +                        if isinstance(node, ast.ImportFrom):
> > +                            if node.module == "__future__":
> > +                                if expected <= set(n.name for n in node.names):
>
> <= is the same as s.issubset(t) (docs here
> <https://docs.python.org/2/library/sets.html#set-objects>). I think it's
> the right check because we want to make sure all the expected future
> imports are present (there can be additional imports too). I'll update the
> code to use the more explicit method instead of the overloaded operator.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/scikit-bio/pull/971/files#r33537569>.
>